src/Core/Core.csproj

```diff
@@ -5,6 +5,7 @@
     <PlatformTarget>x64</PlatformTarget>
     <RootNamespace>Microsoft.Quantum.IQSharp</RootNamespace>
     <AssemblyName>Microsoft.Quantum.IQSharp.Core</AssemblyName>
+    <LangVersion>10.0</LangVersion>
```

Comment:
Question: Why is it required to set the language version to C# 10?

Reply:
The default for `netstandard2.1` is 7.0, IIRC, which doesn't include the `init;` modifier on property access. Added that to make sure that the properties guarded by the new cache aren't accidentally modified post-initialization.